File: java/Stack.java
class Stack {
    private int arr[], top, size;

    Stack(int len) {
        arr = new int[len];
        top = -1;
        size = len;
    }

    public Boolean isFull() {
        return top == size -1;
    }

    public Boolean isEmpty() {
        return top == -1;
    }

    public void push(int num) {
        if (!isFull()) {
            arr[++top] = num;
            System.out.println("inserted:"+num);
            return;
        }
        System.out.println("Stack is full can't insert "+num);
    }

    public int pop() {
        if (!isEmpty()) {
            return arr[top--];
        }
        System.out.println("Stack is Empty");
        return -1;
    }

    public int peek() {
        if (!isEmpty()) {
            return arr[top];
        }
        System.out.println("Stack is Empty");
        return -1;
    }
    
    public void display() {
        if (!isEmpty()) {
            System.out.println("Elements of Stack");
            for (int i = 0; i <= top; i++) {
                System.out.println(arr[i]);
            }
            return;
        }
        System.out.println("Stack is empty nothing to display");
    }

    public static void main(String[] args) {
        Stack s = new Stack(7);
        s.pop();
        s.push(1);
        s.push(2);
        s.push(3);
        s.push(4);
        s.display();
        s.pop();
        s.push(5);
        s.push(6);
        s.push(7);
        s.push(8);
        s.push(0);
        s.pop();
        s.display();
    }
}

File: c++/stack.cpp
#include <iostream>
using namespace std;

#define MAX 7

struct stack {
    int items[MAX];
    int top = -1; // You shouldn't do this
};
typedef struct stack stack;

bool empty(stack &s) {
    return s.top == -1;
}

bool is_full(stack &s) {
    return s.top == MAX - 1;
}

void push(stack &s, int num) {
    if (!is_full(s)) {
        s.items[++s.top] = num;
        return;
    }
    cout << "Stack is full can't insert " << num <<endl;
}

int pop(stack &s) {
    if (!empty(s)) {
        return s.items[s.top--];
    }
    cout << "Stack is empty" <<endl;
    return -1;
}

int peek(stack &s) {
    if (!empty(s)) {
        return s.items[s.top];
    }
    cout << "Stack is empty" <<endl;
    return -1;
}

void display(stack &s) {
    if (!empty(s)) {
        cout << "Elements of stack: " <<endl;
        for (int i = 0; i <= s.top; i++) {
           cout<<s.items[i]<<endl; 
        }
    }
}

int main() {
    stack s;
    pop(s);
    push(s, 1);
    push(s, 2);
    push(s, 3);
    push(s, 4);
    display(s);
    pop(s);
    push(s, 5);
    push(s, 6);
    push(s, 7);
    push(s, 8);
    push(s, 0);
    pop(s);
    display(s);
}
